src/xtgeo/cube/_cube_import.py

```diff
@@ -20,103 +48,105 @@ def import_segy(sfile: xtgeo._XTGeoFile) -> Dict:
+            attributes = _import_segy_all_traces(segyfile)
+    except ValueError as verr:
+        if any([word in str(verr) for word in ["Invalid dimensions", "inconsistent"]]):
+            # cube with missing traces
+            print(
+                "XTGeo notice: Missing or inconsistent traces in SEGY, will try "
+                "but check result carefully!"
+            )
 
-    with segyio.open(sfile, "r") as segyfile:
-        segyfile.mmap()
+            with segyio.open(sfile, "r", ignore_geometry=True) as segyfile:
+                attributes = _import_segy_incomplete_traces(segyfile)
+        else:
+            raise
+    except BaseException as berr:
+        print(f"Cannot parse SEGY file: {str(berr)}")
+        raise
 
-        values = segyio.tools.cube(segyfile)
+    if not attributes:
+        raise RuntimeError("Could not get attributes for segy file")
 
-        logger.info(values.dtype)
-        if values.dtype != np.float32:
-            xtg.warnuser(
-                "Values are converted from {} to {}".format(values.dtype, "float32")
-            )
+    attributes["segyfile"] = sfile
+    return attributes
+
+
+def _import_segy_all_traces(segyfile: segyio.segy.SegyFile) -> Dict:
+    """Import a a full cube SEGY via the SegyIO library to xtgeo format spec.
+
+    Here, the segyio.tools.cube function can be applied
+
+    Args:
+        segyfile: Filehandle from segyio
+    """
+    segyfile.mmap()
+
+    values = _validate_cube_values(segyio.tools.cube(segyfile))
+
+    ilines = segyfile.ilines
+    xlines = segyfile.xlines
+
+    ncol, nrow, nlay = values.shape
+
+    # get additional but required geometries for xtgeo; xori, yori, xinc, ..., rotation
+    attrs = _segy_all_traces_attributes(segyfile, ncol, nrow, nlay)
+
+    attrs["ilines"] = ilines
+    attrs["xlines"] = xlines
+    attrs["values"] = values
+
+    return attrs
+
+
+def _validate_cube_values(values: np.ndarray) -> np.ndarray:
+    """Helper function to validate/check values."""
+    if values.dtype != np.float32:
+        xtg.warnuser(f"Values are converted from {values.dtype} to float32")
+        values = values.astype(np.float32)
+    if np.any(np.isnan(values)):
+        raise ValueError("The input contains NaN values which is trouble!")
+
+    return values
+
+
+def _segy_all_traces_attributes(
+    segyfile: segyio.segy.SegyFile, ncol, nrow, nlay
+) -> Dict:
+    """Get the geometrical values xtgeo needs for a cube definition."""
+    trcode = segyio.TraceField.TraceIdentificationCode
+    traceidcodes = segyfile.attributes(trcode)[:].reshape(ncol, nrow)
+
+    # need positions in corners for making vectors to compute geometries
+    c1v = xcalc.ijk_to_ib(1, 1, 1, ncol, nrow, 1, forder=False)
+    c2v = xcalc.ijk_to_ib(ncol, 1, 1, ncol, nrow, 1, forder=False)
+    c3v = xcalc.ijk_to_ib(1, nrow, 1, ncol, nrow, 1, forder=False)
+
+    xori, yori, zori, zinc = _get_coordinate(segyfile, c1v)
+    cdpx1, cdpy1, _, _ = _get_coordinate(segyfile, c2v)
```

Comment:
Is this common notation? If not, is it possible to use something more descriptive?

Reply:
Yes it is a seismic term `cdp` = common depth point - but I agree; sochcanged to `point_`